### ISSUE TITLE
add --local-fatal-if-no-space

### ISSUE THREAD

**Issue by orena1** [rclone/rclone#8011]
Suggested by @ncw here:
https://forum.rclone.org/t/how-to-make-rclone-copy-local-to-localdestination-stop-when-out-of-space/30859/2?u=orena1

```
We could make a --local-fatal-if-no-space flag or something like that?

That would be fairly easy, except for making it cross platform on all the different platforms rclone supports
```

**Comment by ncw** [comment]
What this requires is working out what error is returned when writing files when the disk gets full on all the platforms rclone supports.

something like `errors.Is(err, syscall.ENOSPC)` will work on unix like platforms and might possibly work on windows.

Using the constant from `x/unix` might be a better idea, so `errors.Is(err, unix.ENOSPC)` but that definitely won't work on windows.

So something like this at the beginning of the Update function

https://github.com/rclone/rclone/blob/705c72d2939a875978d77badad2b56f7e6eb6ca0/backend/local/local.go#L1356

```go
defer func() {
    if err != nil && errors.Is(err, syscall.ENOSPC) {
        err = fserror.sFatalError(err)
    }
}()
```

That needs a config variable and testing on lots of platforms, but that is the basic idea if anyone wants to have a go.

**Comment by ncw** [comment]
I've merged this to master now thanks to @ferrumclaudepilgrim  which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.75 

Please give the --local-fatal-if-no-space flag or the corresponding config file entry a go

**Comment by Dobatymo** [comment]
@ncw I used this flag on rclone 1.75.0 Windows and still got many lines of
```
2026/08/08 01:50:55 ERROR : <path...>: Failed to copy: mkdirat <path...>: There is not enough space on the disk.
2026/08/08 01:50:55 ERROR : <path...>: Not deleting source as copy failed: <path...>: There is not enough space on the disk.
2026/08/08 01:50:55 ERROR : <path...>: Failed to copy: openat <path...>: There is not enough space on the disk.
2026/08/08 01:50:55 ERROR : <path...>: Not deleting source as copy failed: <path...>: There is not enough space on the disk.
```

Using
```
rclone.exe move --checksum --immutable --local-hashes=sha256 --disable=dirmove,move,copy --links --create-empty-src-dirs --delete-empty-src-dirs --local-fatal-if-no-space <path a> <path b>
```

version
> rclone v1.75.0
> - os/version: Microsoft Windows 11 Pro 25H2 25H2 (64 bit)
> - os/kernel: 10.0.26200.8875 (x86_64)
> - os/type: windows
> - os/arch: amd64
> - go/version: go1.26.5
> - go/linking: static
> - go/tags: cmount

**Comment by ncw** [comment]
Thanks for the report @Dobatymo 

Looks like we've left out some windows error codes or something...

Want to have a go a fixing this up @ferrumclaudepilgrim  ?

**Comment by ferrumclaudepilgrim** [comment]
Yes I would. I got started last night and am hoping to have something ready by tonight for review.

**Comment by ferrumclaudepilgrim** [comment]
Whew definitely took longer than I anticipated. Found a second bug on the way in: once detection works on Windows, the VFS cache hits a wait that never returns if the cache cleaner is off. Not new and not Windows-specific, but my change is what gets Windows there. Filed in the same PR. 

That is where the time went. Less writing the fix, more deciding how far to take it. Probably should have pushed earlier for feedback, but once I saw it was a doozy, I wanted to dig all the way through. 

Short version of the root cause: my first test injected ENOSPC, which Windows itself never returns. Sailed through the Windows CI job and told me nothing. That one is on me. Gonna log that one under lessons learned.
